core/src/main/java/com/linecorp/armeria/client/Clients.java

```diff
@@ -182,6 +182,22 @@
      * @see ClientBuilder ClientBuilder, for more information about how the base options and
      *                    additional options are merged when a derived client is created.
      */
+    public static <T> T newDerivedClient(T client,
+                                         ClientBuilderParams currentParams,
```

Comment:
The stub is neither a ClientBuilderParams or proxy. While it'd be possible to create a proxy for the sole purpose of holding ClientBuilderParams, it seems quite heavyweight. Any ideas what might be a good pattern? Perhaps we can also have an SPI StubClientBuilderParamsProvider ?